src/mount.c

```diff
@@ -152,6 +153,17 @@ cc_oci_perform_mount (const struct cc_oci_mount *m, gboolean dry_run)
 	if (dry_run) {
 		return true;
 	}
+	if (g_file_test (m->mnt.mnt_fsname, G_FILE_TEST_IS_REGULAR)) {
+		struct stat st;
+		ret = stat (m->mnt.mnt_fsname, &st);
```

Comment:
You are essentially calling `stat()` twice here: once explicitly and once via `g_file_test()`.

Since you need to `st.st_mode`, Please can you drop the call to `g_file_test()` and just call `stat()` once instead. You can then perform the check using ` S_ISREG(st.st_mode)`.